docs/source/glossary.rst

```diff
@@ -19,21 +19,41 @@ This glossary defines general Sawtooth terms and concepts.
     blockchain is linked by a cryptographic hash to the previous block.
 
   Consensus
-    Process of building agreement among a group of mutually distrusting
-    participants (other nodes on a Sawtooth network). Sawtooth allows
-    different types of consensus on the same blockchain.
-    See also :term:`Dynamic consensus`.
+    Process of reaching agreement among a group of "mutually distrusting"
+    participants (nodes on a Sawtooth network), some of which could be faulty
+    or malicious. Sawtooth supports many types of consensus algorithms. See also
+    :term:`Dynamic consensus`.
+
+  Consensus API
+    Interface that allows a *consensus engine* to interact with the validator
+    in order to handle consensus functionality in a separate process. The
+    Sawtooth consensus API supports a wide variety of consensus algorithms on a
+    Sawtooth network.
+
+  Consensus engine
+    Sawtooth component that provides consensus-specific functionality for a
+    Sawtooth node. The consensus engine runs as a separate process on the node
+    and communicates with the validator through the *consensus API*.
 
   Core
     See :term:`Sawtooth core`.
 
+  Dev mode consensus
+    Simple random-leader consensus algorithm that can be used to test a
+    transaction processor on a single Sawtooth node. (Dev mode is short for
+    "developer mode".) Dev mode cannot be used on a multiple-node network and
```

Comment:
Dev mode can be used on a multiple-node network. it's just very inefficient due to the quantity of forks that are produced and need to be resolved.